src/test/scala/io/citrine/lolo/learners/RandomForestTest.scala

```diff
@@ -161,6 +163,67 @@ class RandomForestTest {
     learner.train(trainingData).getModel()
   }
 
+  def shapleyCompare(
+                      trainingData: Seq[(Vector[Any],Double)],
+                      evalLocation: Vector[Any],
+                      expected: Vector[Double]
+                    ): Unit = {
+    assert(trainingData.length >= 256, "This test is likely to fail randomly when training data are too small.")
+    val actual = RandomForest().train(trainingData).getModel().shapley(evalLocation) match {
+      case None => fail("Unexpected None returned by shapley.")
+      case x: Option[Vector[DenseVector[Double]]] => {
+        val a = x.get
+        assert(a.length == trainingData.head._1.length, "Expected one Shapley value per feature.")
+        assert(a.head.length == 1, "Expected a single output dimension.")
+        a.map{_(0)}
+      }
+      case _ => fail("Unexpected return type.")
+    }
+    expected.zip(actual).foreach {
+      case (e: Double, a: Double) => assert(Math.abs(e - a) < 1)
```

Comment:
Why a tolerance of 1? It seems like it would depend on the problem.

Reply:
Was arbitrary and I hadn't patched it up. Thanks.